src/components/AvroToTypescriptCompiler.ts

```diff
@@ -1,40 +1,68 @@
-import {AvroSchemaInterface} from "../interfaces/AvroSchemaInterface";
-import {AvroSchemaConverter} from "./avroToTypescript/AvroSchemaConverter";
+import {AvroSchemaInterface, RecordType} from "../interfaces/AvroSchemaInterface";
+import { CompilerOutputInterface } from "../interfaces/CompilerOutputInterface";
+import {ExportModel} from "../models/ExportModel";
+import {RecordConverter} from "./avroToTypescript/RecordConverter";
 
 export class AvroToTypescriptCompiler extends BaseCompiler {
     public tsSchemaContent: string;
+    public exports: ExportModel[];
 
-    public async compile(): Promise<void> {
-
-        if (this.isCompileReady() === false) {
-            this.addError(AvroToTypescriptCompiler.errorMessage.notCompileReady);
-            throw new Error(AvroToTypescriptCompiler.errorMessage.notCompileReady);
+    public async compileFolder(): Promise<void> {
+        try {
+            fs.readdir(this.avroSchemaPath, async (err, files) => {
+                for (const file of files) {
+                    try {
+                        const data = fs.readFileSync(this.avroSchemaPath + "/" + file).toString();
+                        await this.compile(JSON.parse(data));
+                    } catch (err) {
+                        console.log(err);
+                    }
+                }
+            });
+        } catch (err) {
+            console.log(err);
         }
 
-        const schemaConverter = new AvroSchemaConverter();
-        const schemaFileHelper = new FileHelper(this.avroSchemaPath);
-        const tsFileHelper = new FileHelper(this.tsSchemaPath);
-        const schemaFileContent: string = (await schemaFileHelper.getContent()).toString();
+        return;
+    }
+
+    /**
+     * @param data - expects data to be JSON parsed
+     * @returns {Promise<CompilerOutputInterface>}
+     */
+    public async compile(data: any): Promise<CompilerOutputInterface> {
+        const recordConverter = new RecordConverter();
+        const recordType: RecordType = data;
 
-        const schemaContent: AvroSchemaInterface = JSON.parse(schemaFileContent) as AvroSchemaInterface;
+        const namespace = recordType.namespace.replace(".", "/");
+        const outputDir = this.tsSchemaPath + namespace;
 
-        if (TypeHelper.isRecordType(schemaContent) === false) {
-            this.addError("Avro schema is not record type");
+        if (TypeHelper.isRecordType(recordType)) {
+            recordConverter.convertRecordToClass(recordType);
+        } else {
+            recordConverter.convertType(recordType);
         }
 
-        if (this.isCompileReady() === false) {
-            throw new Error(AvroToTypescriptCompiler.errorMessage.notCompileReady);
+        const result = recordConverter.joinExports();
+
+        DirHelper.mkdirIfNotExist(outputDir);
+
+        for (const enumFile of recordConverter.enumExports) {
+            fs.writeFileSync(path.resolve(`${outputDir}/${enumFile.name}Enum.ts`), enumFile.content);
```

Comment:
There is a lot of stuff going on in this line. Maybe it would help to extract `path.resolve` to a const. Something like:

```ts
     const basePath = `${outputDir}/${enumFile.name}Enum.ts`;
     const fullPath = path.resolve(basePath);

     fs.writeFileSync(fullPath, enumFile.content);
```
The steps are much clearer when you extract each call

Reply:
Much more readable, thanks!